commit b988f38446d9bfedd2f025e6717bdc348cbd98c6
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sun Nov 15 00:50:58 2015 +0100

    Actually make hg optional (and other plugins while at it)
    
    The existing approach does not work, because if you take KF5WidgetsAddons out
    of the equation, required KF5XmlGui (among others) will be disabled as well.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b793655..6986ffa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,24 +32,18 @@ set_package_properties(DolphinVcs PROPERTIES
     PURPOSE "Provides the version control plugin interface."
 )
 
-find_package(KF5 ${KF5_MIN_VERSION} OPTIONAL_COMPONENTS
-    TextEditor
-    WidgetsAddons
-)
-
 include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDECompilerSettings)
 include(ECMMarkNonGuiExecutable)
+include(ECMOptionalAddSubdirectory)
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-add_subdirectory(svn)
-add_subdirectory(git)
-add_subdirectory(bazaar)
-add_subdirectory(dropbox)
-if (KF5TextEditor_FOUND AND KF5WidgetsAddons_FOUND)
-    add_subdirectory(hg)
-endif()
+ecm_optional_add_subdirectory(svn)
+ecm_optional_add_subdirectory(git)
+ecm_optional_add_subdirectory(bazaar)
+ecm_optional_add_subdirectory(dropbox)
+ecm_optional_add_subdirectory(hg)
 
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git a/hg/CMakeLists.txt b/hg/CMakeLists.txt
index a5e1c7f..fb770e2 100644
--- a/hg/CMakeLists.txt
+++ b/hg/CMakeLists.txt
@@ -1,5 +1,10 @@
 project(fileviewhgplugin)
 
+find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
+    TextEditor
+    WidgetsAddons
+)
+
 add_definitions(-DTRANSLATION_DOMAIN=\"fileviewhgplugin\" -DQT_NO_URL_CAST_FROM_STRING)
 
 set(fileviewhgplugin_SRCS
